src/main/java/net/dries007/tfc/common/recipes/ingredients/HeatableIngredient.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the EUPL, Version 1.2.
+ * You may obtain a copy of the Licence at:
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ */
+
+package net.dries007.tfc.common.recipes.ingredients;
+
+import com.google.gson.JsonObject;
+import net.minecraft.network.FriendlyByteBuf;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.crafting.Ingredient;
+import net.minecraftforge.common.crafting.IIngredientSerializer;
+
+import net.dries007.tfc.common.capabilities.heat.HeatCapability;
+import net.dries007.tfc.common.capabilities.heat.IHeat;
+import net.dries007.tfc.util.JsonHelpers;
+import org.jetbrains.annotations.Nullable;
+
+public class HeatableIngredient extends DelegateIngredient
+{
+    private final int minTemp;
+    private final int maxTemp;
+
+    protected HeatableIngredient(Ingredient delegate, int minTemp, int maxTemp)
+    {
+        super(delegate);
+        this.minTemp = minTemp;
+        this.maxTemp = maxTemp;
+    }
+
+    @Override
+    public IIngredientSerializer<? extends Ingredient> getSerializer()
+    {
+        return Serializer.INSTANCE;
+    }
+
+    @Override
+    public boolean test(@Nullable ItemStack stack)
+    {
+        if (super.test(stack) && stack != null)
+        {
+            return stack.getCapability(HeatCapability.CAPABILITY)
+                .map(IHeat::getTemperature)
+                .filter(temp -> temp > minTemp)
+                .filter(temp -> temp < maxTemp)
+                .isPresent();
```

Comment:
```suggestion
                .map(temp -> temp > minTemp && temp < maxTemp)
                .orElse(false)
```
filter is bad two is worse what idiot wrote this ~~probably me~~